daemon/logger/jsonfile/jsonfile.go

```diff
@@ -1,32 +1,68 @@
 package jsonfile
 
 import (
+	"fmt"
 	"os"
+	"strconv"
 	"sync"
 
 	"github.com/alibaba/pouch/daemon/logger"
+	"github.com/alibaba/pouch/pkg/bytefmt"
 )
 
 // JSONLogFile is uses to log the container's stdout and stderr.
 type JSONLogFile struct {
-	mu sync.Mutex
-
-	f      *os.File
-	perms  os.FileMode
-	closed bool
+	mu          sync.Mutex
+	f           *os.File
+	perms       os.FileMode
+	closed      bool
+	maxSize     uint64 // maximum size of log in byte
+	currentSize uint64 // current size of the latest log in byte
+	maxFile     int    // maximum number of logs
 }
 
 // NewJSONLogFile returns new JSONLogFile instance.
-func NewJSONLogFile(logPath string, perms os.FileMode) (*JSONLogFile, error) {
+func NewJSONLogFile(logPath string, perms os.FileMode, logConfig map[string]string) (*JSONLogFile, error) {
+	var (
+		currentsize uint64 = 0
+		maxsize     uint64 = 0
+		maxfiles    int    = 1
+	)
 	f, err := os.OpenFile(logPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, perms)
 	if err != nil {
 		return nil, err
 	}
 
+	if logConfig != nil {
+		size, err := f.Seek(0, os.SEEK_END)
+		if err != nil {
+			return nil, err
+		}
+		currentsize = uint64(size)
+		if maxsizeString, ok := logConfig["max-size"]; ok {
+			maxsize, err = bytefmt.ToBytes(maxsizeString)
+			if err != nil {
+				return nil, err
+			}
+		}
+		if maxfileString, ok := logConfig["max-file"]; ok {
+			maxfiles, err = strconv.Atoi(maxfileString)
+			if err != nil {
+				return nil, err
+			}
+			if maxfiles < 1 {
+				return nil, fmt.Errorf("max-file cannot be less than 1")
+			}
+		}
+	}
+
 	return &JSONLogFile{
-		f:      f,
-		perms:  perms,
-		closed: false,
```

Comment:
Do we need to add more filed here as https://docs.docker.com/config/containers/logging/json-file/#options, and add a validate func to check in case other filed can't work but appeared in pouch inspect?

Reply:
ok LGTM